src/main/java/eu/darkbot/captcha/CaptchaSolver.java

```diff
@@ -20,106 +22,140 @@
 import java.util.Collections;
 import java.util.Map;
 import java.util.Set;
-import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 public class CaptchaSolver implements CaptchaAPI {
 
     private static final String url = "https://www.darkorbit.com/";
-    private static String key;
-    private static boolean firstTime;
 
-    private static void initFX() {
-        JFXPanel jfxPanel = new JFXPanel();
-        jfxPanel.setPreferredSize(new Dimension(460, 535));
-        Platform.runLater(() -> {
+    @Override
+    public Map<String, String> solveCaptcha(URL url, String frontPage) {
+        if (!url.toString().contains("lp") || !frontPage.contains("recaptcha_image")) return Collections.emptyMap();
+
+        System.out.println("Captcha detected, opening solver...");
+
+        CompletableFuture<String> key = new CompletableFuture<>();
+        SwingUtilities.invokeLater(() -> new SolverJFXPanel(key).display());
+
+        try {
+            String response = key.get(150, TimeUnit.SECONDS); // Worst-case scenario, timeout after 150s
+            if (response != null) return Collections.singletonMap("g-recaptcha-response", response);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            e.printStackTrace();
+        }
+
+        System.out.println("No solution to the captcha was provided, timed out.");
+        return Collections.emptyMap();
+    }
+
+    private static class SolverJFXPanel extends JFXPanel {
+        private final CompletableFuture<String> key;
+
+        private Timeline timeline;
+        private JDialog dialog;
+        private int maxFrames = 120 * 2; // 120 seconds * 2 frames per second
+
+        public SolverJFXPanel(CompletableFuture<String> key) {
+            this.key = key;
+            setPreferredSize(new Dimension(460, 535));
+
+            key.whenComplete((k, t) -> {
+                if (timeline != null) {
+                    timeline.stop();
+                    timeline = null;
+                }
+                if (dialog != null) {
+                    dialog.setVisible(false);
+                    dialog.dispose();
+                    dialog = null;
+                }
+            });
+
+            Platform.runLater(this::prepareWebView);
```

Comment:
```suggestion
            Platform.setImplicitExit(false);
            Platform.runLater(this::prepareWebView);
```

this fix problem of fx thread